src/js/services/request.js

```diff
@@ -27,10 +27,9 @@ angular.module('app')
       // utils
       function setOptions(options) {
         options.headers = options.headers || {};
-        options.timeout = options.timeout || 5000;
+        options.timeout = options.timeout || 20000;
```

Comment:
The dashboard was not working in my environment because of the 5second timeout. Infact, the push request is about adapting the code to work on non-localhost envirnments, where network connections introduce large delays (possibly much larger than 5 seconds) 
BTW, why do you consider 20seconds a high timeout? Our browsers can ask for resources (e.g. web pages or images) and receive responses after tens of seconds without problems (think about mobile phones with slow connections).